code/obj/machinery/dialysis.dm

```diff
@@ -0,0 +1,169 @@
+			new_patient.tri_message(user,\
+			"<span class='notice'><b>[user]</b> begins inserting [src]'s cannulae into [new_patient == user ? "[his_or_her(new_patient)]" : "[new_patient]'s"] arm.</span>",\
+			"<span class='notice'>[new_patient == user ? "You begin" : "<b>[user]</b> begins"] inserting [src]'s cannulae into your arm.</span>",\
+			"<span class='notice'>You begin inserting [src]'s cannulae into [new_patient == user ? "your" : "[new_patient]'s"] arm.</span>")
+			logTheThing(LOG_COMBAT, user, "tries to hook up a dialysis machine [log_reagents(src)] to [constructTarget(new_patient,"combat")] at [log_loc(user)].")
+			SETUP_GENERIC_ACTIONBAR(user, src, 3 SECONDS, PROC_REF(cannulate), list(new_patient, user), src.icon, "dialysis-map", null, null)
+		..()
+
+	process(mult)
+		..()
+		if (!src.patient || !ishuman(src.patient))
+			src.audible_message("<span class='game say'><span class='name'>[src]</span> beeps, \"Patient lost.\"")
+			src.stop_dialysis()
+			return
+
+		if (!src.patient.blood_volume)
+			src.audible_message("<span class='game say'><span class='name'>[src]</span> beeps, \"No blood pressure detected.\"")
+			src.stop_dialysis()
+			return
+
+		if (!in_interact_range(src, src.patient))
+			var/fluff = pick("pulled", "yanked", "ripped")
+			src.patient.visible_message("<span class='alert'><b>[src]'s cannulae gets [fluff] out of [src.patient]'s arm!</b></span>",\
+			"<span class='alert'><b>[src]'s cannulae gets [fluff] out of your arm!</b></span>")
+			src.audible_message("<span class='game say'><span class='name'>[src]</span> beeps, \"No blood pressure detected.\"")
+			src.stop_dialysis()
+			return
+
+		// If we're full of blood and can't take anymore on, destroy it all.
+		if (src.reagents.is_full())
+			src.reagents.clear_reagents()
+			src.audible_message("<span class='game say'><span class='name'>[src]</span> beeps, \"Blood buffer has reached maximum capacity. Purging internal reservoir.\"")
+
+		transfer_blood(src.patient, src, src.draw_amount)
+
+		// Re-implemented here due to all the got dang boutputs.
+		var/list/whitelist_buffer = src.whitelist + src.patient_blood_id
+		for (var/reagent_id in src.reagents.reagent_list)
+			if ((!src.hacked && !whitelist_buffer.Find(reagent_id)) || (src.hacked && whitelist_buffer.Find(reagent_id)))
+				src.reagents.del_reagent(reagent_id)
+
+		src.output_blood_colour = src.reagents.total_volume ? src.reagents.get_average_color().to_rgba() : null
+
+		// Infuse blood back in if possible. Don't wanna stuff too much blood back in.
+		// The blood that's not actually in the bloodstream yet, know what I mean?
+		var/non_blood_volume_blood = src.patient.reagents.reagent_list["blood"]?.volume || 0
+		if (!src.patient.reagents.is_full() && (src.patient.blood_volume + non_blood_volume_blood) <= 500)
+			src.reagents.trans_to(src.patient, src.draw_amount)
+			src.patient.reagents.reaction(src.patient, INGEST, src.draw_amount)
```

Comment:
doesn't an ingest reaction do the transfer anyway?

Reply:
Surprisingly, no. The `trans_to()` line is essential. 😔